### ISSUE TITLE
Job/workflow orchestration

### ISSUE THREAD

**Issue by rasmi** [Recidiviz/recidiviz-data#428]
Ideally, we can use something like Airflow/[Cloud Composer](https://cloud.google.com/composer/) to orchestrate higher-level operations between components of the application.

For example:
- Kick off all the scrape jobs each night.
- Once all the scrape jobs are finished, run the infer release job (#405). (or, run infer_released on each scraper's subset of the data once that scrape is finished).
- Once all of the above is finished, run a data aggregation/dump to BigQuery.
- Once the data dump is finished, update BQ views if necessary and run any custom calculation pipelines.
- Update frontend website data with the fresh results.

**Comment by jessex** [comment]
In general, this sounds very much like the approach I've wanted to take! Glad we're on the same page. I think Cloud Composer sounds like a good fit, but I haven't worked with it or Airflow for a serious project before. I also want us to consider how frequent a certain operation is frequent enough. Let's not pivot to BQ and run all calculation pipelines and queries every day unless we actually need to.

Both of my comments below are about how to trigger from a set of jobs in dynamic windows to another job.

>or, run infer_released on each scraper's subset of the data once that scrape is finished

This was how I was envisioning it in my own mind, for what it's worth. It seems like a more intuitive model to tie this operation to the completion of a particular region, which is apparently our atomic unit of operation.

>Once all of the above is finished, run a data aggregation/dump to BigQuery.

Related to the above comment, knowing that "all of the scrape jobs" finished is non-trivial. Especially once taking into account the likelihood that we'd have different appropriate scrape windows, different expected scrape lengths, and maybe even different appropriate scrape cadences. On one hand, it's probably quite easy to say, "do this only when each of these 100 jobs have run successfully" but you have to provide some sort of window, e.g. "do this only when each of these 100 jobs have run successfully in the past 24 hours," and making a useful, dynamic window may be a challenge. Do you think it's important to be able to do this, or is it sufficient to run this in a more disjoint fashion, for example scheduling a BigQuery pivot for some date and time?

**Comment by arian487** [comment]
Removing this as it is somewhat out of scope for end of fellowship and we are doing some custom orchestration already (for when a job completes)

**Comment by rasmi** [comment]
Ideally we would use Airflow instead cron/queues for BigQuery export and view updating: https://github.com/Recidiviz/recidiviz-data/issues/1311

Using:
* [CloudSqlInstanceExportOperator](https://airflow.apache.org/integration.html#cloudsqlinstanceexportoperator)
* [BigQueryCreateEmptyDatasetOperator](https://airflow.apache.org/integration.html#bigquerycreateemptydatasetoperator)
* [BigQueryCreateExternalTableOperator](https://airflow.apache.org/integration.html#bigquerycreateexternaltableoperator)
* Other [BQ operators](https://airflow.apache.org/integration.html#bigquery-operators) for various checks.
